packages/engine-server/src/store/NoteStore.ts

```diff
@@ -60,6 +60,15 @@ export class NoteStore implements Disposable, INoteStore<string> {
     if (metadata.error) {
       return { error: metadata.error };
     }
+
+    // If note is a stub, return stub note
+    if (metadata.data.stub) {
+      return {
+        data: _.merge(metadata.data, {
```

Comment:
merge will mutate the source object (`metadata.data`). do we want to destructure and return a new object instead?

Reply:
good to know, fixed